### ISSUE TITLE
lint: add zero value re-assignment checker

### ISSUE THREAD

**Issue by Quasilyte** [PieselBois/kfulint#136]
Detect unnecessary zero value assignments.

Should not give a warning for constants that happen to have zero value.

**Before:**
```go
type foo struct {a int; b string}
var f foo
f.a = 0
f.b = ""
```
**After:**
```go
type foo struct {a int; b string}
var f foo
```

**Comment by mihaitodor** [comment]
Would be nice to have such a linter as expressed in the uber-go guidelines [here](https://github.com/uber-go/guide/blob/master/style.md#omit-zero-value-fields-in-structs). Have there been any enhancements added to the framework which would allow implementing such a linter? One issue is that it would likely need to ignore unit test files, since it's probably quite common to have something like the following in test code (copied from the linked guideline):

```go
tests := []struct{
  give string
  want int
}{
  {give: "0", want: 0},
  // ...
}
```
